src/SYNFPAN.m

```diff
@@ -0,0 +1,513 @@
+ . . s vtxt=$get(@json@("entry",SYNZI,"resource","valueCodeableConcept","text"))
+ . . ;i vtxt["Negative" s DHPOBS="Negative" ; 260385009
+ . . i vtxt["Negative" s DHPOBS="Not Detected" ; 260385009
+ . . ;i vtxt["Positive" s DHPOBS="Positive" ; 10828004
+ . . i vtxt["Positive" s DHPOBS="DETECTED" ; 10828004
+ . . i vtxt["Detected" s DHPOBS="DETECTED" ; 260373001
+ . . i vtxt["Not detected" s DHPOBS="Not Detected" ; 260415000
+ . . i vtxt["Confirmed" s DHPOBS="CONFIRMED" ; 
+ . s @eval@("labs",SYNZI,"parms","DHPOBS")=DHPOBS
+ . d log(jlog,"Value is: "_DHPOBS)
+ . ;
+ . i DHPLOINC="33914-3" d  q  ;
+ . . d log(jlog,"Skipping Estimated Glomerular Filtration Rate LOINC: 33914-3")
+ . ;
+ . s DHPUNT=unit
+ . s @eval@("labs",SYNZI,"parms","DHPUNT")=unit
+ . d log(jlog,"Units are: "_unit)
+ . ;
+ . s DHPDTM=hl7time
+ . s @eval@("labs",SYNZI,"parms","DHPDTM")=hl7time
+ . d log(jlog,"HL7 DateTime is: "_hl7time)
+ . ;
+ . s DHPPROV=$$MAP^SYNQLDM("OP","provider")
+ . n DHPPROVIEN s DHPPROVIEN=$o(^VA(200,"B",DHPPROV,""))
+ . if DHPPROVIEN="" S DHPPROVIEN=3
+ . s @eval@("labs",SYNZI,"parms","DHPPROV")=DHPPROVIEN
+ . d log(jlog,"Provider for outpatient is: #"_DHPPROVIEN_" "_DHPPROV)
+ . ;
+ . s DHPLOC=$$MAP^SYNQLDM("OP","location")
+ . n DHPLOCIEN s DHPLOCIEN=$o(^SC("B",DHPLOC,""))
+ . if DHPLOCIEN="" S DHPLOCIEN=4
+ . s @eval@("labs",SYNZI,"parms","DHPLOC")=DHPLOC
+ . d log(jlog,"Location for outpatient is: #"_DHPLOCIEN_" "_DHPLOC)
+ . ;
+ . s @eval@("labs",SYNZI,"status","loadstatus")="readyToLoad"
+ . ;
+ . ;i vistalab="PDW" q  ; skipping because it hangs - gpl wvehr 1/7/21
+ . ;i vistalab="SGPT" q  ; skipping it hangs loinc 1742-6 - gpl wvehr 1/7/21
+ . ;i vistalab="INFLUENZA A RNA" Q  ; likewise
+ . ;i vistalab="INFLUENZA B RNA" Q  ; likewise
+ . ;i vistalab="METAPNEUMOVIRUS RNA" Q  ; likewise
+ . ;
+ q 0
+ ;
+INITMAPS(LOC) ; initialize mapping table for panels
+ ;
+ ; This routine is called by INITMAPS^SYNQLDM - don't run it directly
+ ;
+ N MAP
+ ; vistapanel
+ S MAP="vistapanel"
+ ; Panel type is: 24321-2 Basic metabolic 2000 panel - Serum or Plasma
+ S @LOC@(MAP,"CODE","24321-2","BASIC METABOLIC PANEL")=""
+ ;  Panel type is: 51990-0 Basic metabolic panel - Blood
+ S @LOC@(MAP,"CODE","51990-0","BASIC METABOLIC PANEL")=""
+ ; Panel type is: 24357-6 Urinalysis macro (dipstick) panel - Urine
+ S @LOC@(MAP,"CODE","24357-6","URINALYSIS")=""
+ ; Panel type is: 57698-3 Lipid panel with direct LDL - Serum or Plasma
+ S @LOC@(MAP,"CODE","57698-3","LIPID PROFILE")=""
+ ; Panel type is: 59453-1 Morse Fall Scale panel
+ ;S @LOC@(MAP,"CODE","59453-1","FALLSCALE")=""
+ ; Panel type is: 58410-2 CBC panel - Blood by Automated count
+ S @LOC@(MAP,"CODE","58410-2","CBC")=""
+ ; 
+ S MAP="csample"
+ S @LOC@(MAP,"CODE","24321-2","plasma")=""
+ S @LOC@(MAP,"CODE","24323-8","plasma")=""
+ S @LOC@(MAP,"CODE","57698-3","plasma")=""
+ S @LOC@(MAP,"CODE","24360-0","blood")=""
+ S @LOC@(MAP,"CODE","51990-0","blood")=""
+ S @LOC@(MAP,"CODE","58410-2","blood")=""
```

Comment:
Dead code. We don't use this map. Remove.

Reply:
done